Extensions/TileMap/pixi-tilemap/dist/pixi-tilemap.umd.js

```diff
@@ -1,116 +1,384 @@
+     *
+     * ```
+     * import { CanvasTileRenderer } from '@pixi/tilemap';
+     * import { CanvasRenderer } from '@pixi/canvas-core';
+     *
+     * // You must register this yourself (optional). @pixi/tilemap doesn't do it to
+     * // prevent a hard dependency on @pixi/canvas-core.
+     * CanvasRenderer.registerPlugin('tilemap', CanvasTileRenderer);
+     * ```
+     */
+    // TODO: Move to @pixi/tilemap-canvas
+    class CanvasTileRenderer
+    {
+        /** The renderer */
+        
+
+        /** The global tile animation state */
+        __init() {this.tileAnim = [0, 0];}
+
+        /** @deprecated */
+        __init2() {this.dontUseTransform = false;}
+
+        /** @param renderer */
+        constructor(renderer)
+        {;CanvasTileRenderer.prototype.__init.call(this);CanvasTileRenderer.prototype.__init2.call(this);
             this.renderer = renderer;
             this.tileAnim = [0, 0];
         }
+
+        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+        static getInstance(renderer)
+        {
+            if (!renderer.plugins.tilemap)
+            {
+                renderer.plugins.tilemap = new CanvasTileRenderer(renderer);
+            }
+
+            return renderer.plugins.tilemap;
+        }
     }
-    const cr = PIXI.CanvasRenderer;
-    if (cr) {
-        cr.registerPlugin('tilemap', CanvasTileRenderer);
-    }
 
-    const Constant = {
-        maxTextures: 16,
-        bufferSize: 2048,
-        boundSize: 1024,
-        boundCountPerBuffer: 1,
+    /**
+     * These are additional @pixi/tilemap options.
+     *
+     * This settings should not be changed after the renderer has initialized; otherwise, the behavior
+     * is undefined.
+     */
+    const settings = {
+        /** The default number of textures per tilemap in a tilemap composite. */
+        TEXTURES_PER_TILEMAP: 16,
+
+        /**
+         * The width/height of each texture tile in a {@link TEXTILE_DIMEN}. This is 1024px by default.
+         *
+         * This should fit all tile base-textures; otherwise, {@link TextileResource} may fail to correctly
+         * upload the textures togther in a tiled fashion.
+         */
+        TEXTILE_DIMEN: 1024,
+
+        /**
+         * The number of texture tiles per {@link TextileResource}.
+         *
+         * Texture tiling is disabled by default, and so this is set to `1` by default. If it is set to a
+         * higher value, textures will be uploaded together in a tiled fashion.
+         *
+         * Since {@link TextileResource} is a dual-column format, this should be even for packing
+         * efficiency. The optimal value is usually 4.
+         */
+        TEXTILE_UNITS: 1,
+
+        /** The scaling mode of the combined texture tiling. */
+        TEXTILE_SCALE_MODE: constants.SCALE_MODES.LINEAR,
+
+        /** This will enable 32-bit index buffers. It's useful when you have more than 16K tiles. */
         use32bitIndex: false,
```

Comment:
The attribute already have a documentation.
Should we still add a comment in the code?

Reply:
I'd better be safe than sorry, this helps documenting that we had bugs because we didn't set it to true before